emex_upload/tasks.py

```
@@ -0,0 +1,83 @@
+@app.task(bind=True)
+def insert_emex_data_task(self, clean_file_path: str):
+    """
+    Celery task to insert a clean Emex data file into ClickHouse.
+    """
+    logger.info(f"Starting data insertion for file: {clean_file_path}")
+    self.update_state(state='PROGRESS', meta={'step': 'INSERTING', 'status': 'IN_PROGRESS'})
+    try:
+        if not clean_file_path:
+            raise ValueError("No clean file path provided for insertion.")
+
+        df = pd.read_pickle(clean_file_path)
```

Ruff (0.13.3)

`pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)